Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -218,8 +219,21 @@ void TextureMapperLayer::paintSelf(TextureMapperPaintOptions& options)
         solidColorLayer.setColor(m_state.solidColor);
         contentsLayer = &solidColorLayer;
     }
-    if (!contentsLayer)
+
+    if (!contentsLayer) {
+        // Use the damage information we received from the CoordinatedGraphicsLayer
+        if (!m_damaged.isEmpty()) {
```

Comment:
Do we need to check this? I guess the for loop will not iterate if it's empty

Reply:
done